### ISSUE TITLE
High Load on Server with around ~22K topics and ~220K posts

### ISSUE THREAD

**Issue by cwirsing** [mittwald/typo3-forum#151]
Hi all!

I have a high load problem with typo3_forum: I migrate my old mm_forum data into the database tables of typo3_forum. Everything ist working (list-view, detail-view, pagination etc...) but if i want to add a new topic, my cpus are going 100% till max_execution_time. I could figure out that this happens when $this->topics->attach($topic) is reached at the function addTopic().

At all, i migrated ~22K topics and around 220K Posts to the typo3_forum database tables and with this much data, the typo3_forum steps out at creating a now topic.

If i only migrate 1000 topics and 1000 Posts, creating a topic works by "normal" time and load. I also tested it with 5000 migrated records: It works too but it takes much more (not acceptable) time to add a topic! With 10.000 records: forget it :) 

I have specialists here that have checked the server and the database: Everything OK! We both also checked the apache and the php processes and we could figured out, that there's a maybe a problem with design or extbase by itselfs but we did'nt get further. I think this needs to be tracked deeper. Maybe somebody has some hints on this whole thing? I want to find a solution so everybody can get rid of such problems.

I did'nt say that it is impossible that i did something wrong by migrating the old data but at this point i need feedback on this to continue with finding a solution. Please let me know if you have some ideas or testing results.

All the best,
Christopher

**Comment by pstranghoener** [comment]
Hello, 

i think that is a very special scenario and for that reason i will close the issue. 

Best Regards Philipp

**Comment by martin-helmich** [comment]
Just a quick thought: Extbase's ORM framework is not that efficient when handling large collections, which might also be the issue here. Maybe there is a way refactor the topic creation to work without loading a forum's entire `topics` collection (perhaps dropping the `$forum->addTopic($topic)` call in `Mittwald\Typo3Forum\Domain\Factory\Forum\TopicFactory`)?

**Comment by cwirsing** [comment]
This is exactly what we discovered. We did a further analysis of that "problem" and we have come to the same conclusion.

As mentioned in my first post, we commented out line 612 of file typo3_forum/Classes/Domain/Model/Forum/Forum.php where $this->topics->attach($topic) is coded... this will prevent the errors we have and we can work with the forum "normally".

By the way: Thanks to Stefan Frömken (http://www.typo3lexikon.de/) for spending some time on this! 

All the best,
Chris
